Content.Server/Vehicle/VehicleSystem.cs

```diff
@@ -0,0 +1,253 @@
+
+                component.HasKey = true;
+
+                // Audiovisual feedback
+                _ambientSound.SetAmbience(uid, true);
+            }
+        }
+
+        /// <summary>
+        /// Turn off the engine when key is removed.
+        /// </summary>
+        private void OnEntRemoved(EntityUid uid, VehicleComponent component, EntRemovedFromContainerMessage args)
+        {
+            RemComp<InVehicleComponent>(args.Entity);
+
+            if (_tagSystem.HasTag(args.Entity, "VehicleKey"))
+            {
+                component.HasKey = false;
+                _ambientSound.SetAmbience(uid, false);
+            }
+        }
+
+        /// <summary>
+        /// Depending on which direction the vehicle is facing,
+        /// change its draw depth. Vehicles can choose between special drawdetph
+        /// when facing north or south. East and west are easy.
+        /// </summary>
+        private int GetDrawDepth(TransformComponent xform, bool northOnly)
+        {
+            if (northOnly)
+            {
+                int norfDrawDepth = xform.LocalRotation.Degrees switch
+                {
+                < 135f => 10,
+                <= 225f => 2,
+                _ => 10
+                };
+
+            return norfDrawDepth;
+            }
+            int drawDepth = xform.LocalRotation.Degrees switch
+            {
+              < 45f => 10,
+              <= 315f => 2,
+              _ => 10
+            };
+
+            return drawDepth;
+        }
+
+        /// <summary>
+        /// Change the buckle offset based on what direction the vehicle is facing and
+        /// teleport any buckled entities to it. This is the most crucial part of making
+        /// buckled vehicles work.
+        /// </summary>
+        private void UpdateBuckleOffset(TransformComponent xform, VehicleComponent component)
+        {
+            if (!TryComp<StrapComponent>(component.Owner, out var strap))
+                return;
+            strap.BuckleOffsetUnclamped = xform.LocalRotation.Degrees switch
+            {
+              < 45f => (0, component.SouthOverride),
+              <= 135f => component.BaseBuckleOffset,
+              < 225f  => (0, component.NorthOverride),
+              <= 315f => (component.BaseBuckleOffset.X * -1, component.BaseBuckleOffset.Y),
+              _ => (0, component.SouthOverride)
+            };
+
+            foreach (var buckledEntity in strap.BuckledEntities)
+            {
+                var buckleXform = Transform(buckledEntity);
+                buckleXform.LocalPosition = strap.BuckleOffset;
+            }
+
+        }
+        /// <summary>
+        /// Set the draw depth for the sprite.
+        /// </summary>
+        private void UpdateAppearance(EntityUid uid, int drawDepth)
+        {
+            if (!TryComp<AppearanceComponent>(uid, out var appearance))
+                return;
+
+            appearance.SetData(VehicleVisuals.DrawDepth, drawDepth);
+        }
```

Comment:
this should be renamed `UpdateDrawdepth` if thats the only thing it changes, since theres another method for autoanimate